cli/githubutil.py

```diff
@@ -146,18 +150,110 @@ def list_draft_releases(
     github_cfg_name: str,
     github_repository_owner: str,
     github_repository_name: str,
+    greatest_release_version: CliHint(
+        help=(
+            'Greatest version (according to Semver) up to which (exclusive) '
+            'to consider draft releases, if outdated draft releases are to be determined'
+        ),
+    )=None,
+    list_outdated_draft_releases: bool = False,
 ):
+    '''List all draft releases in a GitHub repository. Optionally, attempt to print only outdated
+    draft releases.
+
+    This function prints a list of names of draft releases found in the given GitHub repository.
+    If the `--list-outdated-draft-releases` flag is set, the printed list is reduced to outdated
+    draft releases smaller than a given release version (according to semver). A draft release is
+    considered outdated iff:
+        1: its version is smaller than the given greatest release version AND
+            2a: it is NOT a hotfix draft release AND
+            2b: there are no hotfix draft releases with the same major and minor version
+            OR
+            3a: it is a hotfix draft release AND
+            3b: there is a hotfix draft release of greater version (according to semver)
+                with the same major and minor version
+
+    Hotfix draft release in this context are draft releases with a semver patch version that is
+    not equal to 0.
+    '''
+    if list_outdated_draft_releases and not greatest_release_version:
+        raise ValueError(
+            "'--greatest-release-version' must be given if '--list-outdated-draft-releases' is set"
+        )
     github_cfg = ctx().cfg_factory().github(github_cfg_name)
     github_helper = GitHubRepositoryHelper(
         owner=github_repository_owner,
         name=github_repository_name,
         github_cfg=github_cfg,
     )
     release_list = [release for release in github_helper.repository.releases() if release.draft]
+    if list_outdated_draft_releases:
+        release_list = _outdated_draft_releases(
+            draft_releases=release_list,
+            greatest_release_version=greatest_release_version,
+        )
     for draft_release in release_list:
         print(draft_release.name)
 
 
+def _outdated_draft_releases(
```

Comment:
i think this function should not reside below `cli/`. the cli module should only contain the required logic to expose cc-utils's API via CLI.
--> move to github/util.py?